src/app/common/components/search-additional-buttons/search-additional-buttons-nav.component.ts

```diff
@@ -5,21 +5,58 @@
  * included in the LICENSE file.
  */
 
-import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from "@angular/core";
-import { SearchAdditionalHeaderButtonInterface } from "./search-additional-buttons.interface";
+import { ChangeDetectionStrategy, Component, EventEmitter, inject, Input, OnInit, Output } from "@angular/core";
+import {
+    SearchAdditionalHeaderButtonInterface,
+    SearchAdditionalHeaderButtonOptions,
+    SearchAdditionalHeaderDatasetInputs,
+} from "./search-additional-buttons.interface";
+import { DatasetBasicsFragment, DatasetKind } from "src/app/api/kamu.graphql.interface";
+import { searchAdditionalButtonsEnum } from "src/app/search/search.interface";
+import { DatasetSubscriptionsService } from "src/app/dataset-view/dataset.subscriptions.service";
+import { Observable, tap } from "rxjs";
 
 @Component({
     selector: "app-search-additional-buttons-nav",
     templateUrl: "./search-additional-buttons-nav.component.html",
     styleUrls: ["./search-additional-buttons.component.scss"],
     changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class SearchAdditionalButtonsNavComponent {
+export class SearchAdditionalButtonsNavComponent implements OnInit {
+    @Input({ required: true }) public datasetBasics: DatasetBasicsFragment;
     @Input({ required: true })
-    public searchAdditionalButtonsData: SearchAdditionalHeaderButtonInterface[];
+    public searchAdditionalButtonsData: SearchAdditionalHeaderButtonOptions;
     @Output() public searchAdditionalButtonsMethod = new EventEmitter<string>();
 
+    private datasetSubsService = inject(DatasetSubscriptionsService);
+    public currentTransformInputs$: Observable<SearchAdditionalHeaderDatasetInputs>;
+
+    public ngOnInit(): void {
+        this.currentTransformInputs$ = this.datasetSubsService.currentTransformInputsCountChanges.pipe(
+            tap((data: SearchAdditionalHeaderDatasetInputs) => {
+                this.searchAdditionalButtonsData[searchAdditionalButtonsEnum.DeriveFrom].counter = data.count;
+            }),
+        );
+    }
+
```

Comment:
Btw, you don't have to write such derived observables in `ngOnInit`. It can be a field, as long as the input observable is declared above it.

Reply:
Removed